internal/test/testdata_test.go

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package test
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/pkg/errors"
+	"go.uber.org/goleak"
+)
+
+func TestGetTestdataPath(t *testing.T) {
+	type args struct {
+		filename string
+	}
+	type want struct {
+		want string
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, string) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got string) error {
+		if !reflect.DeepEqual(got, w.want) {
+			return errors.Errorf("got = %v, want %v", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "get exists test file success",
+			args: args{
+				filename: "tls/dummyCa.pem",
+			},
+			want: want{
+				want: func() string {
+					return baseDir() + "/internal/test/data/tls/dummyCa.pem"
+				}(),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			defer goleak.VerifyNone(tt)
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := GetTestdataPath(test.args.filename)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
+		})
+	}
+}
+
+func Test_baseDir(t *testing.T) {
+	type want struct {
+		want string
+	}
+	type test struct {
+		name       string
+		want       want
+		checkFunc  func(want, string) error
+		beforeFunc func()
+		afterFunc  func()
+	}
+	defaultCheckFunc := func(w want, got string) error {
+		if !reflect.DeepEqual(got, w.want) {
+			return errors.Errorf("got = %v, want %v", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
```

Comment:
Could you please add an abnormal test ?

Reply:
thank you for your comment. Since it only return the file path, I couldn't think any abnormal test....

Reply:
@kevindiu please add error pattern test.

Reply:
I wanna to add it but I couldn't think of any error pattern for testing (as it called `os.Getwd()`. Do you have any idea?